src/Microsoft.Windows.ComputeVirtualization/Container.cs

```diff
@@ -25,22 +24,49 @@ public class ProcessStartInfo
     /// </summary>
     public class Container : IDisposable
     {
-        private string _id;
+        private IntPtr _cs;
         private bool _killOnClose;
         private bool _killed;
 
-        internal Container(string id, bool terminateOnClose)
+        private HcsNotificationWatcher _watcher;
+
+        internal Container(string id, IntPtr computeSystem, bool terminateOnClose)
```

Comment:
This should be a static initialize. Once completed call the actual object constructor right? You are doing an async thread wait in a constructor of an object...